s3pd/__init__.py

```diff
@@ -103,6 +104,38 @@ def download_chunk(bucket, key, shmfileno, offset_first, offset_last, signed):
             Range='bytes=%s-%s' % (offset_first, offset_last))['Body']
         chunk._raw_stream.readinto(shmmap)
 
+def resolve_link(bucket, key, signed, depth=10):
+    # Stop after too many link indirections
+    assert depth > 0, 'Too many levels of link indirections'
+
+    client = create_client(signed)
```

Comment:
Minor: Perhaps I missed something, could we just pass in the same client in the recursion, i.e. resolve_link(bucket, key, client, depth=10)?

Reply:
Fixed in 09a5034